src/js/components/List/index.d.ts

```diff
@@ -18,7 +18,7 @@ export interface ListProps {
   itemProps?: { [_:string]: { background?: string, border?: BorderType, pad?: PadType}};
   margin?: MarginType;
   onMore?: () => void;
-  onClickItem?: (event: React.MouseEvent) => void;
+  onClickItem?: ((event: {item?: {}, index?: number}) => void);
```

Comment:
This will solve the issue, thanks! 
The only thing that may be a problem here is that the event will include more props than item & index and we should be able to allow all keys of the event as part of the TS definition.
for example, if the user will try event.target the typing will fail.

Reply:
@ShimiSun yes you are correct if event.target was used this would throw an error. Also after looking at list docs you can also pass in datum so I will include these two in the TS definitions